Public/Src/Cache/ContentStore/Interfaces/FileSystem/AbsolutePathExtensions.cs

```diff
@@ -20,5 +21,20 @@ public static string GetFileName(this AbsolutePath path)
             Contract.Assume(fileName != null);
             return fileName;
         }
+
+        /// <summary>
+        /// Gets the AbsolutePath for the root of a drive.
+        /// </summary>
+        public static AbsolutePath GetRootPath(this AbsolutePath path)
+        {
+            if (OperatingSystemHelper.IsWindowsOS)
+            {
+                return new AbsolutePath($"{path.DriveLetter}:{Path.DirectorySeparatorChar}");
+            }
+            else
+            {
+                return new AbsolutePath(Path.VolumeSeparatorChar.ToString());
+            }
+        }

```

Comment:
Can you add this to PathTable?

I beleive I have seen this code before in another place...

Reply:
This was in a PR which was abandoned. It doesn't look like it's anywhere else in BuildXL.

---
In reply to: [279586982](https://github.com/Microsoft/BuildXL/pull/220#discussion_r279586982) [](ancestors = 279586982)